packages/realm/src/schema/from-binding.ts

```diff
@@ -16,66 +16,67 @@
 //
 ////////////////////////////////////////////////////////////////////////////
 
-import { PropertyType, TableType } from "../binding";
-import { assert, binding } from "../internal";
-
-// TODO: Update these once the binding expose proper types
-type BindingObjectSchema = binding.Realm["schema"][0];
-type BindingProperty = binding.Realm["schema"][0]["persistedProperties"][0];
+import {
+  ObjectSchema as BindingObjectSchema,
+  Property as BindingProperty,
+  PropertyType as BindingPropertyType,
+  TableType,
+} from "../binding";
```

Comment:
NIT: I like this convention more than `binding.ObjectSchema`.  Perhaps we could re-export somewhere with this naming convention.

Reply:
Good idea, do you mean that you think we should avoid this convention:
```ts
import { binding } from "./internal";
// Use e.g. `Realm`
binding.Realm
```

And only import the things explicitly used in each file:
```ts
import { BindingRealm } from "./internal";
```

Currently we're exporting everything as such:
https://github.com/realm/realm-js/blob/c34f9db228556341776dc9f34bf42aaeb26b1e22/packages/realm/src/internal.ts#L32

But one thing we could do is to modify the generated bindings to already be prefixed with `Binding`. Then we could export it without renaming them (and w/o name conflicts) as such:
```ts
export * from "./binding";
```

Reply:
I like the idea of generating the bindings with the prefix.  Perhaps I'll regret it later, but at the moment it sounds correct, as this would be a good way to deal with a lot of internal naming conflicts.

Reply:
I'll add a ticket for that (I'll add it to the regular backlog rather than the Final v12 epic)